### DISCUSSION TITLE
Victron Battery Life

### DISCUSSION THREAD

**Discussion by iReckyy** [wiggal/GEN24_Ladesteuerung#101] [Ideas]
Hi, was hältst du davon die Funktion Battery Life von Victron zu kopieren/implementieren?

Hebt im Winter bei nicht erreichen von 100% beim SOC die untere SOC Grenze um 5% an um das Voll werden zu gewährleisten.

Wird er voll wird wieder nach unten verstellt.

[Victron Battery Life](https://www.victronenergy.com/media/pg/Energy_Storage_System/de/controlling-depth-of-discharge.html)

Punkt 6.2

**Comment by wiggal** [comment]
Sowas ähnliches gibts schon, allerdings nur in der Modbusvariante.

![grafik](https://github.com/user-attachments/assets/2aa3982b-dc35-41f9-b813-8425baa36a3f)

**Reply by wiggal** [reply]
Die Funktion war mal für den Winter gedacht, um Notstromreserve zu haben, die wollte ich vor dem Winter noch in HTTP implementieren. Mal sehen, ob es bis zum Winter klappt.

**Reply by iReckyy** [reply]
Alles klar.

**Comment by iReckyy** [comment]
Die aktuelle Variante des Notstroms ist schon sehr gut.

Wäre es möglich sie genauso konfigurierbar zu machen wie die frei konfigurierbare Akku-Schonung?

Also bspw.:

Prognose <10KWh = Reserve 15%
Prognose <5KWh = Reserve 20% 
usw.

hat evtl. Zwangsladung zur Folge, vielleicht kann man das irgendwie abfangen.. oder verstellt er nur wenn der aktuelle Stand > Reserve ist?

**Reply by wiggal** [reply]
> Die aktuelle Variante des Notstroms ist schon sehr gut.
> 
> Wäre es möglich sie genauso konfigurierbar zu machen wie die frei konfigurierbare Akku-Schonung?
> 
> Also bspw.:
> 
> Prognose <10KWh = Reserve 15% 
> Prognose <5KWh = Reserve 20% usw.

Schau ich mir bei Gelegenheit mal an.

> hat evtl. Zwangsladung zur Folge, vielleicht kann man das irgendwie abfangen.. oder verstellt er nur wenn der aktuelle Stand > Reserve ist?

Die Notstromreserve wird immer geschrieben, unabhängig vom Ladestand. Hat dann evtl. Zwangsladung zur Folge, was aus meiner Sicht ja sinnvoll ist. Was nützt einem die Funktion, wenn der Strom ausfällt, der Akku leer ist und fast keine PV-Leistung da ist.

**Reply by wiggal** [reply]
OK, hab die freie Konfiguration der Notstromreservierung mal eingebaut.

Hab auch eine Möglichkeit die Zwangsladung zu verhindern eingebaut (Notstromreserve_steuern = 2), musst halt mal probieren, ob das sinnvoll ist. 
Die Option ist nicht dokumentiert!

**Reply by iReckyy** [reply]
Alles klar, werde ich mir anschauen.

Ist das so richtig konfiguriert?

{"5": "30", "10": "25", "15": "20"}

Wann greift die minimal Grenze?

**Reply by wiggal** [reply]
wenn Prognose kleiner 5kWh dann 30%, wenn kleiner 10kWh dann 25%, kleiner 15kWh dann 20%

**Reply by iReckyy** [reply]
Alle klar 👍

**Reply by iReckyy** [reply]
Hi, also irgendwie funktioniert das mit der Notstromgeschichte nicht richtig.

Aktuell habe ich {"5": "30", "6": "28", "7": "26", "8": "24", "9": "22", "10": "20"} drin stehen, aber er stellt mir immer 20% ein, obwohl die Prognose 6,6kWh anzeigt. (also nach der Scikitt-Learn-Optimierung)

Der Schalter steht auf 1, also so, dass er immer verstellen darf.

Habe ich das so richtig eingestellt, oder mache ich was falsch?  Die Reihenfolge der Werte ist egal, oder?

**Reply by wiggal** [reply]
OK, its a Bug
Die Werte lass ich von groß nach klein sortieren, damit die Reigenfolge egal ist, und er sortiert die `10` an den Schluss.

Ich werde es bei Gelegenheit fixen, einstweilen musst du die `10` raus lassen.

Danke für die Info
Wiggal

**Reply by iReckyy** [reply]
Alles klar.
Danke.

**Reply by iReckyy** [reply]
liegt das generell an der 10?  Weil ich habe die generelle Grenze aktuell auch auf 10 stehen.

**Reply by wiggal** [reply]
Es müsste bei allen zweistelligen Zahlen problematisch gewesen sein.
Hab gerade einen FIX im `main` committet.
Siehe #120, könntest du testen.
Danke

**Reply by iReckyy** [reply]
ok, teste ich und gebe morgen Rückmeldung.
